### ISSUE TITLE
CCEI Gaps Analysis: Suitability of CCE infrastructure for current population

### ISSUE THREAD

**Issue by adamdewey** [msupply-foundation/msupply-dashboard#238]
## Background Information 

UNICEF Global have requested a tool to identify gaps in a country's cold storage infrastructure.  

The pain points in this Epic are: 
-  There is no easy way to assess whether the current CCE infrastructure is adequate to serve the current population. 

Problems to solve: 
- A way for central users to view aggregate metrics on the ability of the CCE infrastructure to cater for the demand introduced by immunization programs for the current population
- A way for central users to apply filters to those metrics to narrow down the data they are interested in. 

## Describe the solution you'd like 🎁
**Documents**
- [User stories background](https://docs.google.com/document/d/1Sf7Sj6mCwHTZ0GtEW4O5eLJ9xi7FMHNKnWrHnf8Ueng/edit#heading=h.evuwlqiew02p) 

Mockups: https://docs.google.com/spreadsheets/d/1ow2Jervsw0D51ynHHYGV7GA_XcUIIH_hU-cCSeI0ySc/edit#gid=0

**User stories**
- [ ] C.2.1
  - [ ] Overview of facility cold storage capacity surplus or deficit for each temperature range: +5 °C, -20 °C, -70 °C
- [ ] C.2.2 
  - [ ] Overview of facility CCE hosting capability: On-grid / off-grid 
  - [ ] Overview of facility CCE intervention requirements: No intervention, Replacement, Expansion, Extension
  
#### Filters
- `Facility level` (multi-select)
- `Facility type` (multi-select)
- `Facility ownership` (multi-select)
- `Facility name` (multi-select)

#### Cold storage capacity

![image](https://github.com/msupply-foundation/msupply-dashboard/assets/19318937/1d16a855-8788-4455-a8f6-28ab2140ea81)

**Formula**

- For each temperature storage range (+5 °C, -20 °C and -70 °C):
  - For each supply level:
    - For each facility:

| Displayed | Datapoint  | Formula |
|-| ------------- | ------------- |
|Yes| **Installed capacity**  |  Sum of (`storage volume`) for all CCE instances)  |
|Yes| **Forecast peak demand (litres)** |  ((`Population served` * Total cold storage volume required per capita per year) / `Supply interval`) + Buffer stock  |
|| Total cold storage volume required per capita per year | Vaccine cold storage volume required per capita per year + bundled items cold storage volume required per capita per year |
|| Vaccine cold storage volume required per capita per year | Sum of (volume required for each vaccine course per year per individual * National number of individuals that require that vaccine course per year) for all vaccine courses / (`General population`)  <<- _CAREFUL! Vaccine temperature ranges are not necessarily the same as each other!_ |
|| Volume required for each vaccine course per year per individual | Volume required per dose * number of doses per year / (1 - `Vaccine wastage rate`) |
|| Volume required per dose | Vaccine item packaging volume / Number of doses per packaging |
|| Vaccine item packaging volume | `Packaging type volume` - _Packaging type (Primary, Secondary, Tertiary) dictated by facility's `supply level`_ |
|| Number of doses per packaging | `Packaging type doses` - _Packaging type (Primary, Secondary, Tertiary) dictated by facility's `supply level`_ |
|| Number of doses per year | `Vaccine course total dose number` - _Heuristic: all course doses delivered within 1 year_  |
|| National number of individuals that require that vaccine course per year | `General population` * `Demographic group` |
|| Bundled items cold storage volume required per capita per year | Sum of (volume required for bundled items for each vaccine course per year per individual * National number of individuals that require that vaccine course per year) for all vaccine courses / (`General population`)    <<- _CAREFUL! Vaccine and bundled item temperature ranges are not necessarily the same as each other!_|
|| Volume required for bundled items for each vaccine course per year per individual | Sum of (Vaccine volume required per dose * `Vaccine course total dose number` * `Bundling ratio` * Packaging ratio) for all bundled items per vaccine course|
|| Packaging ratio | `Bundled item packaging type volume` / `Vaccine item packaging type volume`  |
|| Buffer stock | Total cold storage volume required per capita per year * % of year to keep a buffer for * `Population served`  |
|| % of year to keep a buffer for | `Buffer months` / 12 months    |
|Yes| **Forecast surplus / deficit (litres)**  | Installed capacity - Forecast demand  |
|Yes| **Forecast surplus / deficit (%)**  | (Installed capacity - Forecast demand) / Installed capacity  |

### Detail view

TBD

### Moneyworks Jobcode 🧰

CCEIDEV - development
CCEIDSN - design
CCEITEST - testing
CEEIDOC - documentation

**Comment by jmbrunskill** [comment]
Looks great Adam.

Do we need some kind of list of all facilities where forecast demand is greater than expected capacity?

**Comment by adamdewey** [comment]
> Looks great Adam.
> 
> Do we need some kind of list of all facilities where forecast demand is greater than expected capacity?

Yes definitely, sorry - there will be a few graphs needed in each ticket, I'll slowly work my way through adding them!

**Comment by adamdewey** [comment]
@EthanMcQ-TMF - this would be the priority to get in place for the QA team to be able to work against 🙏

**Comment by adamdewey** [comment]
@Mai21 I had a quick compare of the WHO IGA spreadsheet to our dashboard and couldn't arrive at the same numbers. 

Would be good to understand the difference 

![image](https://github.com/user-attachments/assets/08d4dd4b-a4b1-47c1-a07e-5eca229f5e79)

![image](https://github.com/user-attachments/assets/d9f05c85-09fb-4d9c-9ed0-29f1d421dca2)

**Comment by adamdewey** [comment]
This is a copy of the WHO IGA spreadsheet that I've been using: 

[UAT_cce_inventorygapanalysis_tool_2023.xlsx](https://github.com/user-attachments/files/18142514/UAT_cce_inventorygapanalysis_tool_2023.xlsx)

**Comment by Mai21** [comment]
> This is a copy of the WHO IGA spreadsheet that I've been using:
> 
> [UAT_cce_inventorygapanalysis_tool_2023.xlsx](https://github.com/user-attachments/files/18142514/UAT_cce_inventorygapanalysis_tool_2023.xlsx)

@adamdewey I can not find these stores in this excel files. Is it the same file? Sheet is Structures, right?
![Screenshot 2024-12-16 at 1 23 37 PM](https://github.com/user-attachments/assets/c07670b6-026a-479a-ac5d-49ed8db6f179)

![Screenshot 2024-12-16 at 1 24 53 PM](https://github.com/user-attachments/assets/61de53eb-55d5-4807-b395-6a2126a84182)

**Comment by adamdewey** [comment]
@Mai21 - sorry Mai, I attached the wrong version, here is the correct one: 
[UAT_cce_inventorygapanalysis_tool_2023.xlsx](https://github.com/user-attachments/files/18143020/UAT_cce_inventorygapanalysis_tool_2023.xlsx)

**Comment by adamdewey** [comment]
@Mai21 Ok, so I arrived at the same number as WHO IGA using my own workings, see this spreadsheet: 

[GAPS calculation compare.xlsx](https://github.com/user-attachments/files/18145273/GAPS.calculation.compare.xlsx)

The green fields in the spreadsheet have been populated in the GAPS UAT test server (https://uat.msupply.org/)

There is 1 facility (Regional Warehouse 1) that is using 1 vaccine course (Polio bivalent) at -20 °C.

However the dashboard isn't returning the value I would expect based on that data. 

![image](https://github.com/user-attachments/assets/cdd55af5-87c6-4d73-b0a2-3fe2c7ca30e0)

**Comment by adamdewey** [comment]
We now have 2 dashboards designed to meet this need: 

- `3 - Forecasting summary`
- `4 - Forecasting detail`

<img width="1920" height="993" alt="Image" src="https://github.com/user-attachments/assets/f2fcc8a2-c5d4-4677-aa1c-39c0fbd06303" />

<img width="1920" height="946" alt="Image" src="https://github.com/user-attachments/assets/29d98a72-4ea5-4b6b-b25c-56afccbacfc5" />

Closing this issue and will raise any bugs separately
